File: backend/model/post.go
package model

import (
	"github.com/google/uuid"
	"gorm.io/datatypes"
)

type Post struct {
	Model          `gorm:"embedded"`
	Title          string          `json:"title"`
	Description    string          `json:"description"`
	Slug           string          `gorm:"uniqueIndex" json:"slug"`
	Content        string          `gorm:"type:text" json:"content,omitempty"`
	UserId         uuid.UUID       `json:"-"`
	CategoryId     uuid.UUID       `json:"-"`
	Published      *bool           `gorm:"type:boolean;column:published;default:false" json:"published,omitempty"`
	User           *User           `gorm:"foreignKey:UserId" json:"user"`
	Category       *Category       `gorm:"foreignKey:CategoryId" json:"category"`
	Metadata       *datatypes.JSON `json:"metadata,omitempty"`
	SEOTitle       *string         `json:"seo_title,omitempty"`
	SEOKeywords    *string         `json:"seo_keywords,omitempty"`
	SEODescription *string         `json:"seo_description,omitempty"`
}

File: cms/types/post.ts
interface PostUploadData {
  title: string;
  description: string;
  content: string;
  category_id: string;
  published: boolean;
  seo_title: string;
  seo_description: string | null;
  seo_keywords: string | null;
}
interface PostDataTable {
  no: number;
  title: string;
  category: string,
  author: string;
  createdAt: string;
  id: string;
};

interface PostDataDetail {
  id: string, 
  title: string, 
  slug: string,
  description: string,
  author: string,
  content: string,
  published: boolean,
  category_id: string | null,
  seo_title: string | null,
  seo_keywords: string | null,
  seo_description: string | null,
}
export const POST_ENDPOINT = "/v1/system/post.json";
export const POST_DETAIL_ENDPOINT = "/v1/system/post/detail.json";
export type { PostUploadData, PostDataTable, PostDataDetail };
